zebrad/tests/common/lightwalletd/wallet_grpc_test.rs

```diff
@@ -0,0 +1,249 @@
+
+    // Call `GetBlock` with block 1 height
+    let block_one = rpc_client
+        .get_block(BlockId {
+            height: 1,
+            hash: vec![],
+        })
+        .await?
+        .into_inner();
+
+    // Make sure we got block 1 back
+    assert_eq!(block_one.height, 1);
+
+    // Call `GetBlockRange` with the range starting at block 1 up to block 10
+    let mut block_range = rpc_client
+        .get_block_range(BlockRange {
+            start: Some(BlockId {
+                height: 1,
+                hash: vec![],
+            }),
+            end: Some(BlockId {
+                height: 10,
+                hash: vec![],
+            }),
+        })
+        .await?
+        .into_inner();
+
+    // Make sure the returned Stream of blocks is what we expect
+    let mut counter = 0;
+    while let Some(block) = block_range.message().await? {
+        counter += 1;
+        assert_eq!(block.height, counter);
+    }
+
+    // Get the first transction of the first block in the mainnet
+    let hash = zebra_test::vectors::BLOCK_MAINNET_1_BYTES
+        .zcash_deserialize_into::<Block>()
+        .expect("block should deserialize")
+        .transactions[0]
+        .hash()
+        .0
+        .to_vec();
+
+    // Call `GetTransaction` with the transaction hash
+    let transaction = rpc_client
+        .get_transaction(TxFilter {
+            block: None,
+            index: 0,
+            hash,
+        })
+        .await?
+        .into_inner();
+
+    // Check the height of transactions is 1 as expected
+    assert_eq!(transaction.height, 1);
+
+    // TODO: Add after #4216
+    // Currently, this call fails with:
+    // 0: status: Unknown, message: "-32602: Invalid params: invalid length 1, expected a tuple of size 3.", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc"} }
+
+    /*
+    // Call `GetTaddressTxids` with a founders reward address that we know exists and have transactions in the first
+    // few blocks of the mainnet
+    let transactions = rpc_client.get_taddress_txids(TransparentAddressBlockFilter {
+        address: "t3Vz22vK5z2LcKEdg16Yv4FFneEL1zg9ojd".to_string(),
+        range: Some(BlockRange {
+            start: Some(BlockId {
+                height: 1,
+                hash: vec![],
+            }),
+            end: Some(BlockId {
+                height: 10,
+                hash: vec![],
+            }),
+        })
+    }).await?.into_inner();
+
+    dbg!(transactions);
+    */
+
+    // Call `GetTaddressBalance` with a founders reward address
+    let balance = rpc_client
+        .get_taddress_balance(AddressList {
+            addresses: vec!["t3Vz22vK5z2LcKEdg16Yv4FFneEL1zg9ojd".to_string()],
+        })
+        .await?
+        .into_inner();
+
+    // Assume this address will always have a balance greater than zero
+    // TODO: This assumption may be too much, think on something better
```

Comment:
If you call it with a ZFND or Major Grants funding stream address, the balance will always be greater than zero, because new coins are created in each block.

Reply:
Thanks, this is a good idea. The address could change in a future NU and the balance could be zero but it is better than the founders reward one. I made this change at https://github.com/ZcashFoundation/zebra/pull/4253/commits/37bf0d98901b33f733dcca009cf81cdea5ea50a8